batcher/aligned-batcher/src/lib.rs

```diff
@@ -753,43 +615,81 @@ impl Batcher {
     /// Adds verification data to the current batch queue.
     async fn add_to_batch(
         &self,
-        mut batch_state: tokio::sync::MutexGuard<'_, BatchState>,
+        mut batch_state_lock: MutexGuard<'_, BatchState>,
```

Comment:
I believe receiving the mutable reference (`&mut BatchState`) would work and is simpler.

Reply:
That said, it might actually be a great idea to have access to the handle here, since we're sending the error messages here: we should drop the lock before sending the error, otherwise we'll be waiting on the network with the lock taken, not handling other messages in the meantime.

Reply:
Alternatively, it may be cleaner to leave the response to one of the higher layers, and only return errors in the core logic.